src/student_success_tool/analysis/pdp/targets/shared.py

```diff
@@ -280,6 +280,74 @@ def _compute_intensity_num_terms(
         for intensity, time, unit in intensity_time_lefts
     ]
 
+def get_nth_student_terms(
+    df: pd.DataFrame,
+    *,
+    student_id_cols: str | list[str] = "student_guid",
+    sort_cols: str | list[str] = "term_rank",
+    n: int,
+    include_cols: t.Optional[list[str]] = None,
+) -> pd.DataFrame:
+    """
+    For each student, get the nth row in ``df`` (in ascending order of ``sort_cols`` )
+    and a configurable subset of columns. 
+
+    Args:
+        df
+        student_id_cols
+        sort_cols
+        n
+        include_cols
+    """
+    student_id_cols = utils.to_list(student_id_cols)
+    sort_cols = utils.to_list(sort_cols)
+    cols = (
+        df.columns.tolist()
+        if include_cols is None
+        else list(
+            utils.unique_elements_in_order(student_id_cols + sort_cols + include_cols)
+        )
+    )
+    return (
+        df.loc[:, cols]
+        .sort_values(by=sort_cols, ascending=True)
+        .groupby(by=student_id_cols, sort=False, as_index=False)
+        .nth(n)
+    )
+
+
+def get_nth_student_terms_within_cohort(
```

Comment:
I understand that you're repeating a pattern I started elsewhere in the code; however, I think that pattern is meh, and it only happened through historical accident. (The "within-cohort" variation was implemented long after the original.) We can leave this for now, but do ourselves a favor and add a TODO: combine with `get_nth_student_terms()` function by adding within-cohort args to configure behavior.

Reply:
yes agreed!